### ISSUE TITLE
Publicar el corpus de contenido de Telediarios

### ISSUE THREAD

**Issue by dcabo** [civio/verba#96]
En paralelo a #95, que permitirá a cualquiera descargarse sus subtítulos, vamos a publicar el corpus con el que estamos trabajando, para facilitar que otros hagan los análisis que quieran.

Como solución inicial, hasta que sepamos mejor cómo se usa esto, lo más claro parece generar dos corpus: uno con los ficheros de subtítulos originales de la web de RTVE, sin más; y otro con el contenido procesado por nosotros (que ahora mismo es simplemente convertido a frases, porque la extracción de entidades de #10 no está terminada al 100%, así que no la voy a incluir ahora). Voy a generarlos a mano por ahora.

Para generar el primer fichero, `corpus_raw.tar.gz`, en `midas`:

```
$ cd /var/www/verba.civio.es/rtve/staging
$ tar cvzf corpus_raw.tar.gz --exclude=*output* --exclude=*cue* --exclude=*invalid* *vtt *json
$ mv corpus_raw.tar.gz ..
```

Para generar el corpus con el contenido por frases:

```
$ cd /var/www/verba.civio.es/rtve/staging
$ mkdir jq
$ cp *.json jq
$ find *.output.json -exec sh -c "jq -c '{text: .text, start_time: .start_time, end_time: .end_time}' < {} > jq/{}" \;
$ cd jq
$ tar cvzf corpus_cooked.tar.gz --exclude=*cue* *json
$ mv corpus_cooked.tar.gz ../..
$ cd ..
$ rm -rf jq
```

Los ficheros producidos incluyen datos del 23 de diciembre de 2013 hasta el 22 de enero de 2023, inclusive. Vamos a subirlos a [Datos Civio](https://datos.civio.es/), pero de momento los enlaces son:

- Subtítulos originales (VTT) con metadatos de RTVE (JSON): [corpus_raw.tar.gz
](https://www.dropbox.com/s/9wuc60vdet97b80/corpus_raw.tar.gz?dl=0)
- Subtítulos segmentados en frases usando `syntok` (JSON) con metadatos de RTVE (JSON): [corpus_cooked.tar.gz](https://www.dropbox.com/s/esao0k80fs1uusc/corpus_cooked.tar.gz?dl=0)

¿Es este formato cómodo para los que quieren usar los datos?

**Comment by lirondos** [comment]
Pregunta: cada uno de los ficheros del corpus contiene ¿un telediario completo?

**Comment by dcabo** [comment]
@lirondos sí, exacto. En el primero (raw), cada fichero VTT es un Telediario, en el formato original, subtítulos. En el segundo (cooked, perdonad los malos nombres), es el mismo contenido pero partido en frases.

Para cada Telediario hay un fichero con un nombre del estilo 123456.json, que tiene metadatos como el nombre del programa (“Telediario 21h”), fecha, imagen...

Si hay una forma más estándar de organizar esto me decís.

**Comment by davidhguerrero** [comment]
Hola,
he empleado algún tiempo en el preprocesado de los datos y en hacer pruebas con Flair. 

Tengo algunas ideas, muy experimentales todavía, para a empezar a encontrar una manera de solucionar [algunas de estas cuestiones ](https://github.com/civio/verba/issues/12#issuecomment-549311517)

Como he tenido algunos problemas para instalar Flair en local, lo estoy haciendo  con [Kaggel](kaggle.com),. Tiene sus ventajas. 

Si alguno necesita el preprocesado de los datos para ejecutar su código o experimentar con Flair, le comparto el Notebook  =)

Saludos

**Comment by palmerabollo** [comment]
Hola. Los enlaces de dropbox ya no están disponibles y no he encontrado los datos en [Datos Civio](https://datos.civio.es/). ¿Sabéis si están disponibles en algún sitio y cómo importarlos en mi elasticsearch local para cacharrear?

**Comment by dcabo** [comment]
Hola, Guido. Perdona, que no le he podido dedicar nada de tiempo a Verba desde que empezó el virus y lo de los corpus sigue estando cogido con pinzas. Acabo de actualizar los enlaces al principio del issue, apuntando a corpus con datos de hasta ayer. Cualquier cosa me dices.

**Comment by luisignaciomenendez** [comment]
Sin duda! De momento es todo muy preliminar así que aún no puede llamarse proyecto. En cuanto tengamos algo más encauzado os diremos sin falta. Muchas gracias David

**Comment by rubiojr** [comment]
> ¿Es este formato cómodo para los que quieren usar los datos?

Por no marearos pidiendo copias actualizadas, he empezado un pequeño proyectillo (https://github.com/rubiojr/verba-go) para poder tener una copia local de los datos y realizar busquedas locales. Depende de que `verba.civio.es` esté en pié, pero la sincronización con los scripts del repository es eficiente para que solo actualice el contenido nuevo diario.

De momento solo hay algun script para sincronizar los contenidos localmente y poder buscar en una base de datos local, pero llevo idea de publicar una re-implementacion (compatible) del API en Go sin dependencia de Elasticsearch, y publicar el servicio.

Los datos y la idea de Verba, son fantasticos. Gracias por compartir 😃 .

Espero que a alguien le sirva de ayuda 🚀

**Comment by dcabo** [comment]
Pues estupendo, Sergio. Si necesitas cualquier cosa me dices. Verba se nos quedó un poco a medias, porque queríamos añadir funcionalidad en plan _topic modelling_ y nos liamos con otras cosas, pero me alegro de que te sirva. Y eso, cualquier cosa aquí estamos.

**Comment by rubiojr** [comment]
@dcabo esta genial la app, adoro la interface.

De momento estoy cacharreando un poco solo. No veo las noticias 🙈, pero no me importa pasarlas por un LLM y que me de un sumario para empezar el dia 😺. Los transcripts son una fuente de datos que lo facilita. Los sumarios de algunos LLMs salen bastante decentes.

Mi idea es volver a montar el frontend sobre un backend algo menos oneroso, para poder dejarlo en un VPS que me cueste no mas de un par de cafes al mes.

**Comment by dcabo** [comment]
La interfaz es cosa de @carmen-tm, le alegrará oir que te gusta.

Lo de Elastic sí que es un poco grande para arrancar, sí. Tenía sentido porque queríamos hacer más cosas, pero nos quedamos a medias. Pero la gente de Elastic se portó muy bien y nos dio alojamiento en su nube, así que eso que nos ahorramos.

Lo dicho, cualquier cosa aquí estamos.

**Comment by davidhguerrero** [comment]
Hola Sergio,

hace algún tiempo, estructuré los datos y los subí a Kaggle, por si pudiera ser de alguna ayuda =)

https://www.kaggle.com/datasets/davidhguerrero/total-dataset-tve

Aunque no dispongo de mucho tiempo, si necesitas cualquier cosa, también puedes contactarme.

Saludos,

David
________________________________
De: David Cabo ***@***.***>
Enviado: martes, 15 de octubre de 2024 20:47
Para: civio/verba ***@***.***>
Cc: davidhguerrero ***@***.***>; Comment ***@***.***>
Asunto: Re: [civio/verba] Publicar el corpus de contenido de Telediarios (#96)


La interfaz es cosa de @carmen-tm<https://github.com/carmen-tm>, le alegrará oir que te gusta.

Lo de Elastic sí que es un poco grande para arrancar, sí. Tenía sentido porque queríamos hacer más cosas, pero nos quedamos a medias. Pero la gente de Elastic se portó muy bien y nos dio alojamiento en su nube, así que eso que nos ahorramos.

Lo dicho, cualquier cosa aquí estamos.

**Comment by rubiojr** [comment]
Gracias @davidhguerrero 🙇. Al final me decidí por implementar un [pequeño api server](https://github.com/rubiojr/verba-go) que implementa (parte de) el API de Verba y sincroniza el corpus una vez al día desde [verba.civio.es](https://verba.civio.es). Está corriendo en https://verba.rbel.co actualmente 🚀

**Comment by dcabo** [comment]
Cinco años después, he automatizado finalmente la generación de los dos corpus con todo el contenido de Verba. Cada noche a las 4am. [Aquí](https://github.com/civio/verba/blob/master/README.md#corpus-para-descarga) dejo los enlaces permanentes.

😅
